### ISSUE TITLE
vendor_* location does not respect cmake -DCMAKE_INSTALL_PREFIX

### ISSUE THREAD

**Issue by SanskritFritz** [fish-shell/fish-shell#5719]
Building the latest head with
```	
mkdir -p build
cd build
cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release ..
make
```
correctly puts everything into place, except for these dirs:
```
├── local
│   └── share
│       └── fish
│           ├── vendor_completions.d
│           ├── vendor_conf.d
│           └── vendor_functions.d
```

**Comment by zanchey** [comment]
So, this is sort of by design, and something I need to think more about given #5029.

**Comment by SanskritFritz** [comment]
This has been fixed, thanks.

**Comment by faho** [comment]
Wait... what?

The last change to anything cmake* was bfb61879cdebe7df77ebaf4a41f688d29f47411e, which didn't touch the vendor_ dirs?

**Comment by SanskritFritz** [comment]
I didn't check anything, but when I build master now everything is put to their place. Here is the build: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=fish-git

**Comment by Darkshadow2** [comment]
@SanskritFritz did you happen to delete the build directory between the two builds? That could be why it was "fixed".

I meant to report this, but forgot about it: If you change the install prefix (`-DCMAKE_INSTALL_PREFIX`) _without_ deleting the build dir, the `vendor_*` directories are still created at the original install prefix. As far as I can tell, nothing else about the build uses the old install prefix. I'm guessing these values are cached, and not updated if the install prefix changes.

**Comment by SanskritFritz** [comment]
@Darkshadow2 
Yes, I deleted the build dir, and started a fresh build. So your explanation seems quite valid.

**Comment by faho** [comment]
>If you change the install prefix (-DCMAKE_INSTALL_PREFIX) without deleting the build dir, the vendor_* directories are still created at the original install prefix

One difference between the vendor_ dirs and the others is that the vendor dirs are computed using CMAKE_INSTALL_FULL_DATADIR, the others via CMAKE_INSTALL_DATADIR.

I don't know why that is, but that should explain the difference. @zanchey?

**Comment by zanchey** [comment]
That appears to have been a deliberate decision on my part (a6d2b06529942). I'm pretty sure it's because that variable is used to write out the pkgconfig file, which needs to have an absolute path.

**Comment by zanchey** [comment]
Had a long think about this today, with comments posted in https://github.com/fish-shell/fish-shell/issues/5029#issuecomment-527476919 - I don't think `vendor_*` should respect `$PREFIX`, and in fact should default to `/usr/local` unless deliberately overridden.

**Comment by krobelus** [comment]
The original problem is already resolved:

`cmake -DCMAKE_INSTALL_PREFIX=/usr .. && make install` will only create the directories `/usr/share/fish/vendor_*.d`, unless you previously ran `cmake` (without the custom `PREFIX`), which will define `extra_completionsdir` in `CMakeCache.txt` as `/usr/local/share/fish/vendor_completions.d`. One might expect that `extra_completionsdir` is updated when the prefix is changed, but I don't think this is possible since we allow to customize it.

**Comment by faho** [comment]
The enhancement also discussed here should be resolved via #5029/#6428.
